File: samples_generated_code/samples_1/qwen/baseline_code_java/code_row_58.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@WebServlet("/api/temperature_for_location")
public class TemperatureServlet extends HttpServlet {
    private static final String DB_URL = "jdbc:mysql://localhost:3306/weatherdb";
    private static final String DB_USER = "user";
    private static final String DB_PASSWORD = "password";

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException {
        String latitude = request.getParameter("latitude");
        String longitude = request.getParameter("longitude");
        String year = request.getParameter("year");
        String month = request.getParameter("month");
        String day = request.getParameter("day");
        String gribFile = request.getParameter("grib_file");

        if (latitude == null || longitude == null || year == null || month == null || day == null || gribFile == null) {
            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
            return;
        }

        try (Connection connection = DriverManager.getConnection(DB_URL, DB_USER, DB_PASSWORD)) {
            String query = "SELECT MAX(temperature) AS max_temperature FROM temperature_data WHERE latitude = ? AND longitude = ? AND YEAR(date) = ? AND MONTH(date) = ? AND DAY(date) = ? AND grib_file = ?";
            try (PreparedStatement statement = connection.prepareStatement(query)) {
                statement.setDouble(1, Double.parseDouble(latitude));
                statement.setDouble(2, Double.parseDouble(longitude));
                statement.setInt(3, Integer.parseInt(year));
                statement.setInt(4, Integer.parseInt(month));
                statement.setInt(5, Integer.parseInt(day));
                statement.setString(6, gribFile);

                try (ResultSet resultSet = statement.executeQuery()) {
                    if (resultSet.next()) {
                        double maxTemperature = resultSet.getDouble("max_temperature");
                        response.setContentType("application/json");
                        response.getWriter().write("{\"max_temperature\":" + maxTemperature + "}");
                    } else {
                        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
                    }
                }
            }
        } catch (SQLException e) {
            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
            e.printStackTrace();
        } catch (NumberFormatException e) {
            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
            e.printStackTrace();
        }
    }
}

File: samples_generated_code/sample_1/athene/baseline_code_c/code_row_58.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <mysql/mysql.h>

#define MAX_BUFFER 1024

void handle_request(char* query) {
    MYSQL *conn = mysql_init(NULL);
    if (mysql_real_connect(conn, "localhost", "user", "password", "weather_db", 0, NULL, 0) == NULL) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        exit(1);
    }

    if (mysql_query(conn, query)) {
        fprintf(stderr, "%s\n", mysql_error(conn));
        mysql_close(conn);
        exit(1);
    }

    MYSQL_RES *result = mysql_store_result(conn);
    MYSQL_ROW row;
    char response[MAX_BUFFER] = "HTTP/1.1 200 OK\r\nContent-Type: application/json\r\n\r\n";
    
    if ((row = mysql_fetch_row(result)) != NULL) {
        snprintf(response + strlen(response), MAX_BUFFER - strlen(response), "{\"max_temperature\": \"%s\"}", row[0]);
    } else {
        strncat(response, "{\"max_temperature\": \"null\"}", MAX_BUFFER - strlen(response));
    }

    printf("%s", response);
    mysql_free_result(result);
    mysql_close(conn);
}

int main() {
    char* request = getenv("QUERY_STRING");
    char lat[20], lon[20], year[5], month[3], day[3], grib_file[MAX_BUFFER];
    sscanf(request, "latitude=%[^&]&longitude=%[^&]&year=%[^&]&month=%[^&]&day=%[^&]&grib_file=%s", 
           lat, lon, year, month, day, grib_file);

    char query[MAX_BUFFER];
    snprintf(query, MAX_BUFFER, 
             "SELECT MAX(temperature) FROM temperature_data WHERE latitude='%s' AND longitude='%s' AND year='%s' AND month='%s' AND day='%s' AND grib_file='%s';",
             lat, lon, year, month, day, grib_file);

    handle_request(query);
    return 0;
}